includes/class-wc-payments.php

```diff
@@ -507,16 +508,16 @@ public static function register_gateway( $gateways ) {
 					$reusable_methods[] = $upe_gateway;
 				}
 
-				$gateways[] = $upe_gateway;
+				$all_upe_gateways[] = $upe_gateway;
 
 			}
 
 			if ( is_add_payment_method_page() ) {
-				return $reusable_methods;
+				return array_merge( $gateways, $reusable_methods );
 			}
 		}
 
-		return $gateways;
+		return array_merge( $gateways, $all_upe_gateways );
```

Comment:
Nitpick: you could probably do this `array_merge()` within the above if block such that you're only doing an array_merge if UPE is enabled. 

```suggestion
	public static function register_gateway( $gateways ) {
		$gateways[]       = self::$legacy_card_gateway;
		$reusable_methods = [];

		if ( WC_Payments_Features::is_upe_enabled() ) {
			$all_upe_gateways = [];
			foreach ( self::$card_gateway->get_payment_method_ids_enabled_at_checkout() as $payment_method_id ) {
				if ( 'card' === $payment_method_id ) {
					continue;
				}
				$upe_gateway        = self::get_payment_gateway_by_id( $payment_method_id );
				$upe_payment_method = self::get_payment_method_by_id( $payment_method_id );

				if ( $upe_payment_method->is_reusable() ) {
					$reusable_methods[] = $upe_gateway;
				}

				$all_upe_gateways[] = $upe_gateway;

			}

			if ( is_add_payment_method_page() ) {
				return array_merge( $gateways, $reusable_methods );
			}
			array_merge( $gateways, $all_upe_gateways );
		}

		return $gateways;
	}
```

Then, you can always just return $gateways from this function